Sources/IndexStoreDB/IndexStoreDB.swift

```diff
@@ -107,6 +107,26 @@ public final class IndexStoreDB {
     }
   }
 
+  @discardableResult public func forEachRelation(
```

Comment:
This function isn't strictly necessary since the relations attribute exists in SymbolOccurrence, but I figured it would be useful to expose so people can iterate through relations more efficiently?

Reply:
I think it's good to expose the forEach in addition to the method returning an Array, but it should be a method on the occurrence, not on the index.  The difference between finding relations at a specific occurrence, finding all relations of a symbol, and finding all symbols related to a symbol can be kind of confusing, and I think it will help if we put the global queries on the index, and have the simpler "what are the relations of this particular occurrence" on the occurrence itself.